src/frontend/src/hooks/bom.hooks.ts

```diff
@@ -175,6 +176,27 @@ export const useCreateManufacturer = () => {
   );
 };
 
+/**
+ * Custom React hook to delete a material.
+ * @param materialId The material to delete's id
+ * @returns mutation function to delete a material
+ */
+export const useDeleteManufacturer = () => {
+  const queryClient = useQueryClient();
+  return useMutation<any, Error, { manufacturerName: string }>(
+    ['manufacturer', 'delete'],
+    async (payload: { manufacturerName: string }) => {
+      const data = await deleteManufacturer(payload.manufacturerName);
+      return data;
+    },
+    {
+      onSuccess: () => {
+        queryClient.invalidateQueries(['materials', 'manufacturers']);
```

Comment:
we dont care about invalidating the material queries because we have a check in the backend that does not allow us to delete a manufacturer if its associated with a material